lib/ui/window/window.cc

```diff
@@ -13,14 +13,48 @@
 
 namespace flutter {
 
-Window::Window(int64_t window_id, ViewportMetrics metrics)
-    : window_id_(window_id), viewport_metrics_(std::move(metrics)) {
-  library_.Set(tonic::DartState::Current(),
-               Dart_LookupLibrary(tonic::ToDart("dart:ui")));
-}
+Window::Window(tonic::DartPersistentValue& library,
+               int64_t window_id,
+               ViewportMetrics metrics)
+    : library_(library),
+      window_id_(window_id),
+      viewport_metrics_(std::move(metrics)) {}
 
 Window::~Window() {}
 
+void Window::AddView() {
+  std::shared_ptr<tonic::DartState> dart_state = library_.dart_state().lock();
+  if (!dart_state) {
+    return;
+  }
+  tonic::DartState::Scope scope(dart_state);
+  tonic::CheckAndHandleError(tonic::DartInvokeField(
+      library_.value(), "_addView",
```

Comment:
Please cache the field lookup somewhere, see how that's done in `PlatformConfiguration::DidCreateIsolate`.